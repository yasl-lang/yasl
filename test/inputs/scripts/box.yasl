
let box = undef

const boxmt = {
    .tostr: (fn(self) -> "box(#{self.contents->tostr()})"),
    .__bnot: (fn(self) -> self.contents),
    .update: fn(self, val) {
        self.contents = val
    },
    .__set: mt.get({}).__set
}

box = fn(val) {
    const tmp = { .contents: val }
    mt.set(tmp, boxmt)
    return tmp
}


const x = box(10)

echo x
echo ^x

x->update("blah blah blah")

echo x
echo ^x


