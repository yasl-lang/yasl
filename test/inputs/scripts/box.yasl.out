box(10)
10
box(blah blah blah)
blah blah blah
